### PR TITLE
View capybara

### PR DESCRIPTION
**Pull request by iambenkis**
## Description
In this project, We create integration tests for all the views used in my project. 

### Checkout  :1st_place_medal: 

- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view :heavy_check_mark: 
-  Use Capybara to write integration tests for each view in your project. :heavy_check_mark: 
  
- User index page: :heavy_check_mark: 
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page: :heavy_check_mark: 
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page: :heavy_check_mark: 
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page: :heavy_check_mark: 
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commenter.
  - I can see the comment each commenter left.

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      expect(page).to have_content(@post.comments_counter)
+    end
+
+    it 'I can see how many likes a post has.' do
+      expect(page).to have_content(@post.likes_counter)
+    end
+
+    it 'When I click on a post, it redirects me to that post show page.' do
+      click_link(@post.title)
+      expect(current_path).to match user_post_path(user_id: @subject1.id, id: @post.id)
```
Comment (Davidosky007) [inline spec/features/post_index_spec.rb]: * Kindly write test to cover this spec `I can see a section for pagination if there are more posts than fit on the view.`

File: app/models/post.rb [hunk 2]
```diff
@@ -14,6 +14,6 @@ def update_posts_counter
   end
 
   def recent_comments
-    comments.order(created_at: :desc).limit(5)
+    comments.includes(:author).order(created_at: :desc).limit(5)
```
Comment (Davidosky007) [inline app/models/post.rb]: * Kindly solve the N + 1 problem  when fetching all posts and their comments for a user, you can do this in your posts controller index action [Check requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @iambenkis 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Good looking readme file 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (peterrobert) [reply spec/features/post_index_spec.rb]: This change has been done well

File: app/models/post.rb (hunk 2 above)
Comment (peterrobert) [reply app/models/post.rb]: This change has been done well

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi @iambenkis, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️

### OPTIONAL

- [ ] It will be great to remove the Microverse logo from your readme file. This will make your project more professional. It was supposed to be requested by the first reviewer but he missed it. So I am making it optional.


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**